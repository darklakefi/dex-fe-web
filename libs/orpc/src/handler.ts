import { RPCHandler } from "@orpc/server/fetch";
import { CORSPlugin } from "@orpc/server/plugins";
import { router } from "./router";

export const handler = new RPCHandler(router, {
  plugins: [
    new CORSPlugin({
      origin: (origin) => origin,
      allowMethods: ["GET", "HEAD", "PUT", "POST", "DELETE", "PATCH"],
    }),
  ],
});
