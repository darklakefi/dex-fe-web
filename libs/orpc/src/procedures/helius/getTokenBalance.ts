import z from "zod";
import { baseProcedure } from "../baseProcedure";

const getTokenBalanceInputSchema = z.object({
  address: z.string(),
  ownerAddress: z.string(),
});

export const getTokenBalance = baseProcedure
  .input(getTokenBalanceInputSchema)
  .handler(async ({ input }) => {
    // return await searchAssetsHandler({ input });
  });
