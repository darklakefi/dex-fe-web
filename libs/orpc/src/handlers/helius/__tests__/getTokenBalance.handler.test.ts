import { randUuid } from "@ngneat/falso";
import { getTokenBalanceHandler } from "../getTokenBalance.handler";

describe("getTokenBalanceHandler", () => {
  const ownerAddress = randUuid();

  it("should return a list of assets", async () => {
    const result = await getTokenBalanceHandler({
      ownerAddress,
    });

    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
  });
});
