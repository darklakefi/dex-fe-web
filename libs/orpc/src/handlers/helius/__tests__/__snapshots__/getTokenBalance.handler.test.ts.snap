// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getTokenBalanceHandler > should return a list of assets 1`] = `
{
  "assets": [
    {
      "description": "New ABC 13 9370, 13.3, 5th Gen CoreA5-8250U, 8GB RAM, 256GB SSD, power UHD Graphics, OS 10 Home, OS Office A & J 2016",
      "id": "0cf84d21-ea2c-4dbe-8336-85d544e7281d",
      "image": null,
      "name": "Practical Granite Keyboard",
      "symbol": "TAS",
    },
    {
      "description": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
      "id": "dcaa170f-2189-4745-a6ce-d47a41ef27d0",
      "image": null,
      "name": "Licensed Concrete Fish",
      "symbol": "AWE",
    },
    {
      "description": "The automobile layout consists of a front-engine design, with transaxle-type transmissions mounted at the rear of the engine and four wheel drive",
      "id": "766eb696-4d4e-44c9-8af8-db0d72705666",
      "image": null,
      "name": "Tasty Concrete Chips",
      "symbol": "INC",
    },
    {
      "description": "The Football Is Good For Training And Recreational Purposes",
      "id": "0f1a2f72-5144-462d-bbda-abef1b73a50e",
      "image": null,
      "name": "Generic Metal Shirt",
      "symbol": "SMA",
    },
    {
      "description": "Bostons most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
      "id": "e5963c7f-dfc9-48ec-b396-92b271f33a44",
      "image": null,
      "name": "Refined Fresh Shirt",
      "symbol": "FAN",
    },
    {
      "description": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
      "id": "9fb722ac-277f-439b-8be0-058186974372",
      "image": null,
      "name": "Licensed Metal Chips",
      "symbol": "TAS",
    },
    {
      "description": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
      "id": "c083920a-55d9-4061-aeb6-20e956558735",
      "image": null,
      "name": "Awesome Rubber Salad",
      "symbol": "TAS",
    },
    {
      "description": "Bostons most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
      "id": "6b50fd9e-d078-4395-9834-d6be653cfe42",
      "image": null,
      "name": "Rustic Concrete Chips",
      "symbol": "HAN",
    },
    {
      "description": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
      "id": "a0c6789a-30eb-4cfd-9a73-0a4ebaa497a4",
      "image": null,
      "name": "Handcrafted Wooden Gloves",
      "symbol": "RUS",
    },
    {
      "description": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
      "id": "750537d9-a1e4-4696-892c-68cb6df0f25c",
      "image": null,
      "name": "Unbranded Granite Chicken",
      "symbol": "TAS",
    },
  ],
  "ownerAddress": "341a6c55-d6f5-4ab5-867b-b0f159cb49bb",
  "tokenAccounts": [
    {
      "address": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
      "amount": 906466,
      "mint": "17f4a3c0-badc-4a17-8f21-89174526ced4",
    },
    {
      "address": "4cb6932d-b540-4464-b66e-b6964d4e74c9",
      "amount": 290423,
      "mint": "4bd821d2-a605-4715-b3ca-0f1a2f725144",
    },
    {
      "address": "f688dd29-19c4-4be5-b1e9-096ec3e5963c",
      "amount": 498798,
      "mint": "32821048-e405-4c14-a468-15a5dd445f9f",
    },
    {
      "address": "c78e650c-793c-43f9-ad3e-7e59d645c6c2",
      "amount": 741975,
      "mint": "352c0bb8-ab6d-4f46-8dd8-554d99d4a1a1",
    },
    {
      "address": "653cfe42-e62a-4d68-8c64-f507e3384624",
      "amount": 599623,
      "mint": "0a4ebaa4-97a4-4fc8-a90e-70ae24c17ece",
    },
    {
      "address": "96092c68-cb6d-40f2-9cbe-e2b393adfc17",
      "amount": 418322,
      "mint": "5b85dafd-df63-4696-8b02-5ed0ca7bbed7",
    },
    {
      "address": "0b110e8f-87b9-4990-8de4-0ec34c74a091",
      "amount": 155566,
      "mint": "b4622e48-1afa-44ae-be94-d19be7b39336",
    },
    {
      "address": "0e82c196-ce91-4fb4-86dd-be60ac6bce66",
      "amount": 525454,
      "mint": "85645e43-bd3f-49df-885d-8eee9e89d595",
    },
    {
      "address": "71ff1785-8ddb-4314-bb89-f4ecf04cf20e",
      "amount": 612304,
      "mint": "9bac24ee-d28d-4fd1-9727-ff524106e523",
    },
    {
      "address": "c82d3f3c-4a40-45ac-8da9-76c09db29be3",
      "amount": 4038,
      "mint": "ac1cd6bc-f3e1-4ab7-82ba-2d25cf63e63d",
    },
  ],
  "total": 10,
}
`;
